From 8717de78231531244a792a3f497e2b663cd75db9 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 21 Jun 2018 23:11:44 +0000
Subject: [PATCH] Add Clear Linux docker integration for kata containers

Add Kata Containers as a runtime option for Docker and make it
the default.

Signed-off-by: Daniela Plascencia <daniela.plascencia@linux.intel.com>
---
 50-runtime.conf            | 2 ++
 clearlinux.conf            | 8 ++++++++
 docker-set-runtime.service | 8 ++++++++
 set-docker-default-runtime | 5 +++++
 4 files changed, 23 insertions(+)
 create mode 100644 50-runtime.conf
 create mode 100644 clearlinux.conf
 create mode 100644 docker-set-runtime.service
 create mode 100755 set-docker-default-runtime

diff --git a/50-runtime.conf b/50-runtime.conf
new file mode 100644
index 0000000..e05cf06
--- /dev/null
+++ b/50-runtime.conf
@@ -0,0 +1,2 @@
+[Service]
+Environment="DOCKER_DEFAULT_RUNTIME=--default-runtime runc"
diff --git a/clearlinux.conf b/clearlinux.conf
new file mode 100644
index 0000000..6587392
--- /dev/null
+++ b/clearlinux.conf
@@ -0,0 +1,8 @@
+[Unit]
+Wants=docker-set-runtime.service
+After=docker-set-runtime.service
+
+[Service]
+Environment="DOCKER_EXTRA_RUNTIMES=--add-runtime cc-runtime=/usr/bin/cc-runtime --add-runtime kata-runtime=/usr/bin/kata-runtime"
+ExecStart=
+ExecStart=/usr/bin/dockerd $DOCKER_EXTRA_RUNTIMES $DOCKER_DEFAULT_RUNTIME --storage-driver=overlay2
diff --git a/docker-set-runtime.service b/docker-set-runtime.service
new file mode 100644
index 0000000..0cc4fdd
--- /dev/null
+++ b/docker-set-runtime.service
@@ -0,0 +1,8 @@
+[Unit]
+Description=Set Default Docker Runtime
+ConditionPathExists=!/etc/systemd/system/docker.service.d/50-runtime.conf
+ConditionPathExists=/dev/kvm
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/set-docker-default-runtime
diff --git a/set-docker-default-runtime b/set-docker-default-runtime
new file mode 100755
index 0000000..970522e
--- /dev/null
+++ b/set-docker-default-runtime
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+mkdir -p /etc/systemd/system/docker.service.d/
+echo -e '[Service]\nEnvironment="DOCKER_DEFAULT_RUNTIME=--default-runtime kata-runtime"' > /etc/systemd/system/docker.service.d/50-runtime.conf
+systemctl daemon-reload
-- 
2.18.0

